packages/builders/src/components/ActionRow.ts

```diff
@@ -38,17 +39,17 @@ export class ActionRowBuilder<T extends AnyComponentBuilder> extends ComponentBu
 	 * @param components The components to add to this action row.
 	 * @returns
 	 */
-	public addComponents(components: T[]) {
-		this.components.push(...components);
+	public addComponents(...components: RestOrArray<T>) {
+		this.components.push(...normalizeArray(...components));
 		return this;
 	}
 
 	/**
 	 * Sets the components in this action row
 	 * @param components The components to set this row to
 	 */
-	public setComponents(components: T[]) {
-		this.components.splice(0, this.components.length, ...components);
+	public setComponents(...components: RestOrArray<T>) {
+		this.components.splice(0, this.components.length, ...normalizeArray(...components));
```

Comment:
Instead of spreading here again you can make normalizeArray to accept array input.

Reply:
It does I think? I just couldn’t figure out a way to make TS happy with this without doing this but if you know how then please suggest on the function itself